spec/lib/webauthn/registration_service_spec.rb

```diff
@@ -142,10 +147,83 @@
     end
 
     it "raises a MalformedAttestationError" do
-      expect { service.register_second_factor_security_key }.to raise_error(
+      expect { service.register_security_key }.to raise_error(
         DiscourseWebauthn::MalformedAttestationError,
         I18n.t("webauthn.validation.malformed_attestation_error"),
       )
     end
   end
+
+  context "when the user presence flag is false" do
+    it "raises a UserPresenceError" do
+      # simulate missing user presence by flipping first bit to 0
+      flags = "00000010"
+      overridenAuthData = service.send(:auth_data)
+      overridenAuthData[32] = [flags].pack("b*")
+
+      service.instance_variable_set(:@auth_data, overridenAuthData)
+
+      expect { service.register_security_key }.to raise_error(
+        DiscourseWebauthn::UserPresenceError,
+        I18n.t("webauthn.validation.user_presence_error"),
+      )
+    end
+  end
+
+  it "works" do
```

Comment:
This description is too generic IMO, I think it'll be helpful to describe what exactly we are testing for.